commonmark_extensions/tables.py

```diff
@@ -107,6 +117,9 @@ def finalize(parser=None, block=None):
             if len(list(filter(lambda cell : not re.match(r"[-=:]+$", cell), row))) == 0:
                 # This row has cells of just dahses.
                 if len(table_parts) == 1:
+                    if len(row) != len(table_parts[0][0]):
+                        block.t = 'paragraph'
```

Comment:
As mentioned in commit message, this doesn't actually change the Python type of the block, but it does produce the right output on rendering so likely doesn't matter.